lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -2875,5 +2876,57 @@ def test_cell_method_correct_order(self):
         self.assertTrue(cube1 == cube2)
 
 
+class Test__dimensional_metadata:
+    @pytest.fixture
+    def cube(self):
+        return stock.simple_2d_w_cell_measure_ancil_var()
+
+    def test_not_found(self, cube):
+        with pytest.raises(KeyError, match="was not found in"):
+            cube._dimensional_metadata("grid_latitude")
+
+    def test_dim_coord_name_found(self, cube):
+        res = cube._dimensional_metadata("bar")
+        assert res == cube.coord("bar")
+
+    def test_dim_coord_instance_found(self, cube):
+        res = cube._dimensional_metadata(cube.coord("bar"))
+        assert res == cube.coord("bar")
+
+    def test_aux_coord_name_found(self, cube):
+        res = cube._dimensional_metadata("wibble")
+        assert res == cube.coord("wibble")
+
+    def test_aux_coord_instance_found(self, cube):
+        res = cube._dimensional_metadata(cube.coord("wibble"))
+        assert res == cube.coord("wibble")
+
+    def test_cell_measure_name_found(self, cube):
+        res = cube._dimensional_metadata("cell_area")
+        assert res == cube.cell_measure("cell_area")
+
+    def test_cell_measure_instance_found(self, cube):
+        res = cube._dimensional_metadata(cube.cell_measure("cell_area"))
+        assert res == cube.cell_measure("cell_area")
+
+    def test_ancillary_var_name_found(self, cube):
+        res = cube._dimensional_metadata("quality_flag")
+        assert res == cube.ancillary_variable("quality_flag")
+
+    def test_ancillary_var_instance_found(self, cube):
+        res = cube._dimensional_metadata(
+            cube.ancillary_variable("quality_flag")
+        )
+        assert res == cube.ancillary_variable("quality_flag")
+
+    def test_two_with_same_name(self, cube):
+        # If a cube has two _DimensionalMetadata objects with the same name, the
+        # current behaviour results in _dimensional_metadata returning the first
+        # one it finds
+        cube.cell_measure("cell_area").rename("wibble")
+        res = cube._dimensional_metadata("wibble")
+        assert res == cube.coord("wibble")
```

Comment:
It might be useful to also test the case for 
```res = cube._dimensional_metadata(cube.cell_measure("wibble"))```
just to ensure it's possible to specify this dimensional metadata by being more particular.